src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -288,7 +288,7 @@ private void ConsumeFromRemainingBuffer(int bytesToConsume)
             {
                 await WriteAsciiStringAsync(HttpKnownHeaderNames.Cookie).ConfigureAwait(false);
                 await WriteTwoBytesAsync((byte)':', (byte)' ').ConfigureAwait(false);
-                await WriteAsciiStringAsync(cookiesFromContainer).ConfigureAwait(false);
+                await WriteStringAsync(cookiesFromContainer).ConfigureAwait(false);
```

Comment:
Potentially another call site that was missed? Line 249: `await WriteAsciiStringAsync(header.Key.Name).ConfigureAwait(false);`

Reply:
That should be fine; the header names are validated by the headers collection, either on their way in, or if validation wasn't enabled then, while enumerating them on the way out.  I don't believe we could end up with a header here if the name contained invalid characters.

Reply:
>@davidsh, am I wrong? Could we end up with non-ASCII characters in a header name of a header enumerated from an HttpHeaders collection?

The HttpHeaders class is supposed to prevent non-ASCII characters from being used in a headername.  And sending such a headername on the wire (presumably using UTF-8 encoding of the octets of the Unicode in the headername) would be a violation of the RFC for HTTP.

If we have a test that shows that we're doing this, then it would be a bug.  But it would be a bug that needs to get fixed in HttpHeaders and not in the SocketsHttpHandler.

So, it's fine to use WriteStringAsync here.

Reply:
Thanks, David.

Reply:
See: https://tools.ietf.org/html/rfc7230#page-22.  But in general, 'token', is simply alphanumeric and no Unicode chars.

```
     header-field   = field-name ":" OWS field-value OWS

     field-name     = token

...
     token          = 1*tchar

     tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
                    / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
                    / DIGIT / ALPHA
                    ; any VCHAR, except delimiters
```